plugins/inputs/http_listener/http_listener.go

```diff
@@ -111,25 +112,35 @@ func (t *HttpListener) httpListen() error {
 func (t *HttpListener) ServeHTTP(res http.ResponseWriter, req *http.Request) {
 	t.wg.Add(1)
 	defer t.wg.Done()
-	body, err := ioutil.ReadAll(req.Body)
-	if err != nil {
-		log.Printf("Problem reading request: [%s], Error: %s\n", string(body), err)
-		http.Error(res, "ERROR reading request", http.StatusInternalServerError)
-		return
-	}
 
 	switch req.URL.Path {
 	case "/write":
-		var metrics []telegraf.Metric
-		metrics, err = t.parser.Parse(body)
-		if err == nil {
-			for _, m := range metrics {
-				t.acc.AddFields(m.Name(), m.Fields(), m.Tags(), m.Time())
+		var http400 bool
+		var http400msg string
+		var partial string
+		scanner := bufio.NewScanner(req.Body)
+		for scanner.Scan() {
+			metrics, err := t.parser.Parse(scanner.Bytes())
+			if err == nil {
+				for _, m := range metrics {
+					t.acc.AddFields(m.Name(), m.Fields(), m.Tags(), m.Time())
+				}
+				partial = "partial write: "
+			} else {
+				http400msg += err.Error() + " "
```

Comment:
It does need to build up the error, because the expected behavior is to parse & write the well-formed messages. I can change it to use a `bytes.Buffer` :+1: